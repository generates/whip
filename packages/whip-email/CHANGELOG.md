# @generates/whip-email

## 0.0.2

### Patch Changes

- aec0aff: Adding default transport host

## 0.0.1

### Patch Changes

- 500a0df: Initial release
