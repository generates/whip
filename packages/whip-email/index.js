import nodemailer from 'nodemailer'
// import Mailgen from 'mailgen'

export default function redisPlugin (app, opts) {
  app.nodemailer = nodemailer.createTransport(opts.transport)
  // app.mailgen = new Mailgen(opts.mailgen)
  app.use(function redisMiddleware (req, res, next) {
    req.nodemailer = app.nodemailer
    // req.mailgen = app.mailgen
    next()
  })
}
