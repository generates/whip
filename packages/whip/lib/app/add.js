export default async function add (...plugins) {
  for (const { plugin, opts } of plugins) {
    // Execute the plugin with the app (this) and any passed options.
    const returned = plugin(this, opts)

    // If the plugin function is a promise, wait for it to resolve.
    if (returned?.then) await returned
  }
}
