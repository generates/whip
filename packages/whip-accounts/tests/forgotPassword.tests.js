import { test } from '@ianwalter/bff'
import { getTestEmail, extractToken } from '@generates/whip-email'
import app from './fixtures/app.js'
import { accounts } from './fixtures/accounts.js'

const generalUser = accounts.find(a => a.firstName === 'General')
const disabledUser = accounts.find(a => a.firstName === 'Disabled')

test('Forgot Password • Invalid emails', async t => {
  let res = await app.test('/forgot-password').post({ email: null })
  t.expect(res.statusCode).toBe(400)
  t.expect(res.body).toMatchSnapshot()

  res = await app.test('/forgot-password').post({ email: '' })
  t.expect(res.statusCode).toBe(400)
  t.expect(res.body).toMatchSnapshot()

  const email = 'babu_frik@example'
  res = await app.test('/forgot-password').post({ email })
  t.expect(res.statusCode).toBe(400)
  t.expect(res.body).toMatchSnapshot()
})

test('Forgot Password • Unregistered email', async t => {
  const email = 'babu_frik@example.com'
  const res = await app.test('/forgot-password').post({ email })
  t.expect(res.statusCode).toBe(200)
  t.expect(res.body).toEqual({
    message: 'Forgot Password request submitted successfully'
  })
})

test('Forgot Password • Registered email', async t => {
  const res = await app.test('/forgot-password').post(generalUser)
  t.expect(res.statusCode).toBe(200)
  t.expect(res.body).toEqual({
    message: 'Forgot Password request submitted successfully'
  })

  await t.asleep(1000)

  // Extract and verify the Forgot Password email and token.
  const byEmail = email => email.headers.to === generalUser.email
  const { email } = await extractToken(byEmail)
  t.expect(email).toMatchSnapshot({
    id: t.expect.any(String),
    messageId: t.expect.any(String),
    source: t.expect.any(String),
    date: t.expect.any(String),
    time: t.expect.any(String),
    envelope: {
      remoteAddress: t.expect.any(String)
    },
    headers: t.expect.any(Object)
  })

  // Verify the Forgot Password token was inserted into the database.
  const where = { email: generalUser.email, type: 'password' }
  const record = await app.prisma.token.findFirst({ where })

  t.expect(record).toBeDefined()
})

test('Forgot Password • Disabled user', async t => {
  const res = await app.test('/forgot-password').post(disabledUser)
  t.expect(res.statusCode).toBe(200)
  t.expect(res.body).toEqual({
    message: 'Forgot Password request submitted successfully'
  })

  // Verify no email was sent to the user.
  await t.asleep(1000)
  const email = await getTestEmail(e => e.headers.to === disabledUser.email)
  t.expect(email).toBe(undefined)
})
