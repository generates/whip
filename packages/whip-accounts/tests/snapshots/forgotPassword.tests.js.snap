// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot Password • Invalid emails 1`] = `
Object {
  "feedback": Object {
    "email": Array [
      "A valid email is required.",
    ],
  },
  "message": "Validation Error",
}
`;

exports[`Forgot Password • Invalid emails 2`] = `
Object {
  "feedback": Object {
    "email": Array [
      "A valid email is required.",
    ],
  },
  "message": "Validation Error",
}
`;

exports[`Forgot Password • Invalid emails 3`] = `
Object {
  "feedback": Object {
    "email": Array [
      "A valid email is required.",
    ],
  },
  "message": "Validation Error",
}
`;

exports[`Forgot Password • Registered email 1`] = `
Object {
  "date": Any<String>,
  "envelope": Object {
    "from": Object {
      "address": "",
      "args": false,
    },
    "host": "[127.0.0.1]",
    "remoteAddress": Any<String>,
    "to": Array [
      Object {
        "address": "general_test@example.com",
        "args": false,
      },
    ],
  },
  "headers": Any<Object>,
  "html": "
    <!doctype html>
    <html xmlns=\\"http://www.w3.org/1999/xhtml\\" xmlns:v=\\"urn:schemas-microsoft-com:vml\\" xmlns:o=\\"urn:schemas-microsoft-com:office:office\\">
      <head>
        <title>
          
        </title>
        <!--[if !mso]><!-->
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
        <!--<![endif]-->
        <meta http-equiv=\\"content-type\\" content=\\"text/html; charset=utf-8\\" />
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <style type=\\"text/css\\">
          #outlook a { padding:0; }
          body { margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%; }
          table, td { border-collapse:collapse;mso-table-lspace:0pt;mso-table-rspace:0pt; }
          img { border:0;height:auto;line-height:100%; outline:none;text-decoration:none;-ms-interpolation-mode:bicubic; }
          p { display:block;margin:13px 0; }
        </style>
        <!--[if mso]>
        <noscript>
        <xml>
        <o:OfficeDocumentSettings>
          <o:AllowPNG/>
          <o:PixelsPerInch>96</o:PixelsPerInch>
        </o:OfficeDocumentSettings>
        </xml>
        </noscript>
        <![endif]-->
        <!--[if lte mso 11]>
        <style type=\\"text/css\\">
          .mj-outlook-group-fix { width:100% !important; }
        </style>
        <![endif]-->
        
      <!--[if !mso]><!-->
        <link href=\\"https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700\\" rel=\\"stylesheet\\" type=\\"text/css\\">
        <style type=\\"text/css\\">
          @import url(https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700);
        </style>
      <!--<![endif]-->

    
        
    <style type=\\"text/css\\">
      @media only screen and (min-width:480px) {
        .mj-column-per-100 { width:100% !important; max-width: 100%; }
      }
    </style>
    <style media=\\"screen and (min-width:480px)\\">
      .moz-text-html .mj-column-per-100 { width:100% !important; max-width: 100%; }
    </style>
    
  
        <style type=\\"text/css\\">
        
        
        </style>
        <style type=\\"text/css\\">
        a {
        color: #7870FA;
        font-weight: 500;
      }
      .wrapper {
        padding: 32px;
      }
      .social-icon {
        display: inline-block;
        padding: 0 20px;
      }
      @media (max-width: 414px) {
        .wrapper {
          padding: 0;
        }
        .social-icon {
          display: inline-block;
          padding: 0 8px;
        }
      }
        </style>
        
      </head>
      <body style=\\"word-spacing:normal;background-color:#E9EAEB;\\">
        
        
      <div
         style=\\"background-color:#E9EAEB;\\"
      >
        
      
      <!--[if mso | IE]><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"wrapper-outlook\\" role=\\"presentation\\" style=\\"width:600px;\\" width=\\"600\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->
    
      
      <div  class=\\"wrapper\\" style=\\"margin:0px auto;max-width:600px;\\">
        
        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:0;text-align:center;\\"
              >
                <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" width=\\"600px\\" ><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" role=\\"presentation\\" style=\\"width:600px;\\" width=\\"600\\" bgcolor=\\"#F7F7F7\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->
    
      
      <div  style=\\"background:#F7F7F7;background-color:#F7F7F7;margin:0px auto;max-width:600px;\\">
        
        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"background:#F7F7F7;background-color:#F7F7F7;width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:12px 32px;text-align:center;\\"
              >
                <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr></tr></table><![endif]-->
              </td>
            </tr>
          </tbody>
        </table>
        
      </div>
    
      
      <!--[if mso | IE]></td></tr></table></td></tr><tr><td class=\\"\\" width=\\"600px\\" ><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" role=\\"presentation\\" style=\\"width:600px;\\" width=\\"600\\" bgcolor=\\"#F7F7F7\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->
    
      
      <div  style=\\"background:#F7F7F7;background-color:#F7F7F7;margin:0px auto;max-width:600px;\\">
        
        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"background:#F7F7F7;background-color:#F7F7F7;width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:12px 32px;text-align:center;\\"
              >
                <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" style=\\"vertical-align:top;width:536px;\\" ><![endif]-->
            
      <div
         class=\\"mj-column-per-100 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"
      >
        
      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"
      >
        <tbody>
          
              <tr>
                <td
                   style=\\"font-size:0px;word-break:break-word;\\"
                >
                  
      <div
         style=\\"height:28px;line-height:28px;\\"
      >&#8202;</div>
    
                </td>
              </tr>
            
              <tr>
                <td
                   align=\\"left\\" style=\\"font-size:0px;padding:0;word-break:break-word;\\"
                >
                  
      <div
         style=\\"font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:28px;font-weight:500;line-height:150%;text-align:left;color:#221E12;\\"
      >Reset your password</div>
    
                </td>
              </tr>
            
              <tr>
                <td
                   style=\\"font-size:0px;word-break:break-word;\\"
                >
                  
      <div
         style=\\"height:16px;line-height:16px;\\"
      >&#8202;</div>
    
                </td>
              </tr>
            
              <tr>
                <td
                   align=\\"left\\" style=\\"font-size:0px;padding:0;word-break:break-word;\\"
                >
                  
      <div
         style=\\"font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:16px;line-height:150%;text-align:left;color:#5C5A52;\\"
      >To get started, please click the button below:</div>
    
                </td>
              </tr>
            
              <tr>
                <td
                   style=\\"font-size:0px;word-break:break-word;\\"
                >
                  
      <div
         style=\\"height:16px;line-height:16px;\\"
      >&#8202;</div>
    
                </td>
              </tr>
            
              <tr>
                <td
                   align=\\"left\\" vertical-align=\\"middle\\" style=\\"font-size:0px;padding:0;word-break:break-word;\\"
                >
                  
      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"border-collapse:separate;line-height:100%;\\"
      >
        <tbody>
          <tr>
            <td
               align=\\"center\\" bgcolor=\\"#7870FA\\" role=\\"presentation\\" style=\\"border:none;border-radius:3px;cursor:auto;mso-padding-alt:12px 16px;background:#7870FA;\\" valign=\\"middle\\"
            >
              <a
                 href=\\"http://localhost/reset-password?email=general_test@example.com&token=YgIHAYzWQic0T-8h6ZF6aE2j9YWwEYokcbnzdcrbC0XW3xF2iUYoxymNjqXwcP49\\" style=\\"display:inline-block;background:#7870FA;color:white;font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:16px;font-weight:normal;line-height:120%;margin:0;text-decoration:none;text-transform:none;padding:12px 16px;mso-padding-alt:0px;border-radius:3px;\\" target=\\"_blank\\"
              >
                Reset
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    
                </td>
              </tr>
            
              <tr>
                <td
                   style=\\"font-size:0px;word-break:break-word;\\"
                >
                  
      <div
         style=\\"height:28px;line-height:28px;\\"
      >&#8202;</div>
    
                </td>
              </tr>
            
        </tbody>
      </table>
    
      </div>
    
          <!--[if mso | IE]></td></tr></table><![endif]-->
              </td>
            </tr>
          </tbody>
        </table>
        
      </div>
    
      
      <!--[if mso | IE]></td></tr></table></td></tr><tr><td class=\\"\\" width=\\"600px\\" ><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" role=\\"presentation\\" style=\\"width:600px;\\" width=\\"600\\" bgcolor=\\"white\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->
    
      
      <div  style=\\"background:white;background-color:white;margin:0px auto;max-width:600px;\\">
        
        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"background:white;background-color:white;width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:12px 12px;text-align:center;\\"
              >
                <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" style=\\"vertical-align:top;width:576px;\\" ><![endif]-->
            
      <div
         class=\\"mj-column-per-100 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"
      >
        
      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"
      >
        <tbody>
          
              <tr>
                <td
                   style=\\"font-size:0px;word-break:break-word;\\"
                >
                  
      <div
         style=\\"height:64px;line-height:64px;\\"
      >&#8202;</div>
    
                </td>
              </tr>
            
              <tr>
                <td
                   align=\\"center\\" style=\\"font-size:0px;padding:0;word-break:break-word;\\"
                >
                  
      <div
         style=\\"font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:13px;line-height:150%;text-align:center;color:#5C5A52;\\"
      ></div>
    
                </td>
              </tr>
            
        </tbody>
      </table>
    
      </div>
    
          <!--[if mso | IE]></td></tr></table><![endif]-->
              </td>
            </tr>
          </tbody>
        </table>
        
      </div>
    
      
      <!--[if mso | IE]></td></tr></table></td></tr><tr><td class=\\"\\" width=\\"600px\\" ><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" role=\\"presentation\\" style=\\"width:600px;\\" width=\\"600\\" bgcolor=\\"white\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->
    
      
      <div  style=\\"background:white;background-color:white;margin:0px auto;max-width:600px;\\">
        
        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"background:white;background-color:white;width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:12px 32px;text-align:center;\\"
              >
                <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" style=\\"vertical-align:top;width:536px;\\" ><![endif]-->
            
      <div
         class=\\"mj-column-per-100 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"
      >
        
      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"
      >
        <tbody>
          
              <tr>
                <td
                   style=\\"font-size:0px;word-break:break-word;\\"
                >
                  
      <div
         style=\\"height:32px;line-height:32px;\\"
      >&#8202;</div>
    
                </td>
              </tr>
            
        </tbody>
      </table>
    
      </div>
    
          <!--[if mso | IE]></td></tr></table><![endif]-->
              </td>
            </tr>
          </tbody>
        </table>
        
      </div>
    
      
      <!--[if mso | IE]></td></tr></table></td></tr></table><![endif]-->
              </td>
            </tr>
          </tbody>
        </table>
        
      </div>
    
      
      <!--[if mso | IE]></td></tr></table><![endif]-->
    
    
      </div>
    
      </body>
    </html>
  
",
  "id": Any<String>,
  "messageId": Any<String>,
  "priority": "normal",
  "read": false,
  "source": Any<String>,
  "subject": "Reset your password",
  "time": Any<String>,
  "to": Array [
    Object {
      "address": "general_test@example.com",
      "name": "",
    },
  ],
}
`;

exports[`Forgot Password • Registered email 2`] = `
Object {
  "date": Any<String>,
  "envelope": Object {
    "from": Object {
      "address": "",
      "args": false,
    },
    "host": "[127.0.0.1]",
    "remoteAddress": Any<String>,
    "to": Array [
      Object {
        "address": "general_test@example.com",
        "args": false,
      },
    ],
  },
  "headers": Any<Object>,
  "html": "
    <!doctype html>
    <html xmlns=\\"http://www.w3.org/1999/xhtml\\" xmlns:v=\\"urn:schemas-microsoft-com:vml\\" xmlns:o=\\"urn:schemas-microsoft-com:office:office\\">
      <head>
        <title>

        </title>
        <!--[if !mso]><!-->
        <meta http-equiv=\\"X-UA-Compatible\\" content=\\"IE=edge\\">
        <!--<![endif]-->
        <meta http-equiv=\\"content-type\\" content=\\"text/html; charset=utf-8\\" />
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
        <style type=\\"text/css\\">
          #outlook a { padding:0; }
          body { margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%; }
          table, td { border-collapse:collapse;mso-table-lspace:0pt;mso-table-rspace:0pt; }
          img { border:0;height:auto;line-height:100%; outline:none;text-decoration:none;-ms-interpolation-mode:bicubic; }
          p { display:block;margin:13px 0; }
        </style>
        <!--[if mso]>
        <noscript>
        <xml>
        <o:OfficeDocumentSettings>
          <o:AllowPNG/>
          <o:PixelsPerInch>96</o:PixelsPerInch>
        </o:OfficeDocumentSettings>
        </xml>
        </noscript>
        <![endif]-->
        <!--[if lte mso 11]>
        <style type=\\"text/css\\">
          .mj-outlook-group-fix { width:100% !important; }
        </style>
        <![endif]-->

      <!--[if !mso]><!-->
        <link href=\\"https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700\\" rel=\\"stylesheet\\" type=\\"text/css\\">
        <style type=\\"text/css\\">
          @import url(https://fonts.googleapis.com/css?family=Ubuntu:300,400,500,700);
        </style>
      <!--<![endif]-->



    <style type=\\"text/css\\">
      @media only screen and (min-width:480px) {
        .mj-column-per-100 { width:100% !important; max-width: 100%; }
      }
    </style>
    <style media=\\"screen and (min-width:480px)\\">
      .moz-text-html .mj-column-per-100 { width:100% !important; max-width: 100%; }
    </style>


        <style type=\\"text/css\\">


        </style>
        <style type=\\"text/css\\">
        a {
        color: #7870FA;
        font-weight: 500;
      }
      .wrapper {
        padding: 32px;
      }
      .social-icon {
        display: inline-block;
        padding: 0 20px;
      }
      @media (max-width: 414px) {
        .wrapper {
          padding: 0;
        }
        .social-icon {
          display: inline-block;
          padding: 0 8px;
        }
      }
        </style>

      </head>
      <body style=\\"word-spacing:normal;background-color:#E9EAEB;\\">


      <div
         style=\\"background-color:#E9EAEB;\\"
      >


      <!--[if mso | IE]><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"wrapper-outlook\\" role=\\"presentation\\" style=\\"width:600px;\\" width=\\"600\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->


      <div  class=\\"wrapper\\" style=\\"margin:0px auto;max-width:600px;\\">

        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:0;text-align:center;\\"
              >
                <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" width=\\"600px\\" ><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" role=\\"presentation\\" style=\\"width:600px;\\" width=\\"600\\" bgcolor=\\"#F7F7F7\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->


      <div  style=\\"background:#F7F7F7;background-color:#F7F7F7;margin:0px auto;max-width:600px;\\">

        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"background:#F7F7F7;background-color:#F7F7F7;width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:12px 32px;text-align:center;\\"
              >
                <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr></tr></table><![endif]-->
              </td>
            </tr>
          </tbody>
        </table>

      </div>


      <!--[if mso | IE]></td></tr></table></td></tr><tr><td class=\\"\\" width=\\"600px\\" ><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" role=\\"presentation\\" style=\\"width:600px;\\" width=\\"600\\" bgcolor=\\"#F7F7F7\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->


      <div  style=\\"background:#F7F7F7;background-color:#F7F7F7;margin:0px auto;max-width:600px;\\">

        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"background:#F7F7F7;background-color:#F7F7F7;width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:12px 32px;text-align:center;\\"
              >
                <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" style=\\"vertical-align:top;width:536px;\\" ><![endif]-->

      <div
         class=\\"mj-column-per-100 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"
      >

      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"
      >
        <tbody>

              <tr>
                <td
                   style=\\"font-size:0px;word-break:break-word;\\"
                >

      <div
         style=\\"height:28px;line-height:28px;\\"
      >&#8202;</div>

                </td>
              </tr>

              <tr>
                <td
                   align=\\"left\\" style=\\"font-size:0px;padding:0;word-break:break-word;\\"
                >

      <div
         style=\\"font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:28px;font-weight:500;line-height:150%;text-align:left;color:#221E12;\\"
      >Reset your password</div>

                </td>
              </tr>

              <tr>
                <td
                   style=\\"font-size:0px;word-break:break-word;\\"
                >

      <div
         style=\\"height:16px;line-height:16px;\\"
      >&#8202;</div>

                </td>
              </tr>

              <tr>
                <td
                   align=\\"left\\" style=\\"font-size:0px;padding:0;word-break:break-word;\\"
                >

      <div
         style=\\"font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:16px;line-height:150%;text-align:left;color:#5C5A52;\\"
      >To get started, please click the button below:</div>

                </td>
              </tr>

              <tr>
                <td
                   style=\\"font-size:0px;word-break:break-word;\\"
                >

      <div
         style=\\"height:16px;line-height:16px;\\"
      >&#8202;</div>

                </td>
              </tr>

              <tr>
                <td
                   align=\\"left\\" vertical-align=\\"middle\\" style=\\"font-size:0px;padding:0;word-break:break-word;\\"
                >

      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"border-collapse:separate;line-height:100%;\\"
      >
        <tbody>
          <tr>
            <td
               align=\\"center\\" bgcolor=\\"#7870FA\\" role=\\"presentation\\" style=\\"border:none;border-radius:3px;cursor:auto;mso-padding-alt:12px 16px;background:#7870FA;\\" valign=\\"middle\\"
            >
              <a
                 href=\\"http://localhost/reset-password?email=general_test@example.com&token=NNEqWDjRfU5OABZEq0YXFom6ruguRKNelSd_Nkq6IqsfYruFqny6uzCrvf9k1c2U\\" style=\\"display:inline-block;background:#7870FA;color:white;font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:16px;font-weight:normal;line-height:120%;margin:0;text-decoration:none;text-transform:none;padding:12px 16px;mso-padding-alt:0px;border-radius:3px;\\" target=\\"_blank\\"
              >
                Reset
              </a>
            </td>
          </tr>
        </tbody>
      </table>

                </td>
              </tr>

              <tr>
                <td
                   style=\\"font-size:0px;word-break:break-word;\\"
                >

      <div
         style=\\"height:28px;line-height:28px;\\"
      >&#8202;</div>

                </td>
              </tr>

        </tbody>
      </table>

      </div>

          <!--[if mso | IE]></td></tr></table><![endif]-->
              </td>
            </tr>
          </tbody>
        </table>

      </div>


      <!--[if mso | IE]></td></tr></table></td></tr><tr><td class=\\"\\" width=\\"600px\\" ><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" role=\\"presentation\\" style=\\"width:600px;\\" width=\\"600\\" bgcolor=\\"white\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->


      <div  style=\\"background:white;background-color:white;margin:0px auto;max-width:600px;\\">

        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"background:white;background-color:white;width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:12px 12px;text-align:center;\\"
              >
                <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" style=\\"vertical-align:top;width:576px;\\" ><![endif]-->

      <div
         class=\\"mj-column-per-100 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"
      >

      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"
      >
        <tbody>

              <tr>
                <td
                   style=\\"font-size:0px;word-break:break-word;\\"
                >

      <div
         style=\\"height:64px;line-height:64px;\\"
      >&#8202;</div>

                </td>
              </tr>

              <tr>
                <td
                   align=\\"center\\" style=\\"font-size:0px;padding:0;word-break:break-word;\\"
                >

      <div
         style=\\"font-family:Ubuntu, Helvetica, Arial, sans-serif;font-size:13px;line-height:150%;text-align:center;color:#5C5A52;\\"
      ></div>

                </td>
              </tr>

        </tbody>
      </table>

      </div>

          <!--[if mso | IE]></td></tr></table><![endif]-->
              </td>
            </tr>
          </tbody>
        </table>

      </div>


      <!--[if mso | IE]></td></tr></table></td></tr><tr><td class=\\"\\" width=\\"600px\\" ><table align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" class=\\"\\" role=\\"presentation\\" style=\\"width:600px;\\" width=\\"600\\" bgcolor=\\"white\\" ><tr><td style=\\"line-height:0px;font-size:0px;mso-line-height-rule:exactly;\\"><![endif]-->


      <div  style=\\"background:white;background-color:white;margin:0px auto;max-width:600px;\\">

        <table
           align=\\"center\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"background:white;background-color:white;width:100%;\\"
        >
          <tbody>
            <tr>
              <td
                 style=\\"direction:ltr;font-size:0px;padding:12px 32px;text-align:center;\\"
              >
                <!--[if mso | IE]><table role=\\"presentation\\" border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\"><tr><td class=\\"\\" style=\\"vertical-align:top;width:536px;\\" ><![endif]-->

      <div
         class=\\"mj-column-per-100 mj-outlook-group-fix\\" style=\\"font-size:0px;text-align:left;direction:ltr;display:inline-block;vertical-align:top;width:100%;\\"
      >

      <table
         border=\\"0\\" cellpadding=\\"0\\" cellspacing=\\"0\\" role=\\"presentation\\" style=\\"vertical-align:top;\\" width=\\"100%\\"
      >
        <tbody>

              <tr>
                <td
                   style=\\"font-size:0px;word-break:break-word;\\"
                >

      <div
         style=\\"height:32px;line-height:32px;\\"
      >&#8202;</div>

                </td>
              </tr>

        </tbody>
      </table>

      </div>

          <!--[if mso | IE]></td></tr></table><![endif]-->
              </td>
            </tr>
          </tbody>
        </table>

      </div>


      <!--[if mso | IE]></td></tr></table></td></tr></table><![endif]-->
              </td>
            </tr>
          </tbody>
        </table>

      </div>


      <!--[if mso | IE]></td></tr></table><![endif]-->


      </div>

      </body>
    </html>

",
  "id": Any<String>,
  "messageId": Any<String>,
  "priority": "normal",
  "read": false,
  "source": Any<String>,
  "subject": "Reset your password",
  "time": Any<String>,
  "to": Array [
    Object {
      "address": "general_test@example.com",
      "name": "",
    },
  ],
}
`;
